From d2def279fbf2c5279adf028493f79b60b225ddcb Mon Sep 17 00:00:00 2001
From: Vinith Kumar R <quic_vinithku@quicinc.com>
Date: Mon, 7 Oct 2024 23:46:43 +0530
Subject: [PATCH] wifi: ath12k: Handle NULL check before accessing arvif

In ath12k_mac_get_arvif_by_vdev_id function call, NULL value may
return. In this case, need to avoid accessing the arvif and exit
the ath12k_htt_vdev_txrx_stats_handler function to avoid crash.

Signed-off-by: Vinith Kumar R <quic_vinithku@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_rx.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/dp_rx.c b/drivers/net/wireless/ath/ath12k/dp_rx.c
index db3a4a1..1fca49f 100644
--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -2310,6 +2310,11 @@ static void ath12k_htt_vdev_txrx_stats_handler(struct ath12k_base *ab,
 			vdev_id = __le32_to_cpu(vdev_tlv->vdev_id);
 			arvif = ath12k_mac_get_arvif_by_vdev_id(ab, vdev_id);
 
+			if(!arvif){
+				rcu_read_unlock();
+				return;
+			}
+
 			arvif->vdev_stats.rx_msdu_byte_cnt =
 				HTT_VDEV_GET_STATS_U64(vdev_tlv->rx_msdu_byte_cnt_lo,
 						       vdev_tlv->rx_msdu_byte_cnt_hi);
-- 
2.34.1

